docs/reference/capabilities.md

```diff
@@ -27,6 +27,7 @@ about capabilities, refer to the [Appium documentation](https://appium.io/docs/e
 | <div style="width:10em">Capability</div> | Description |
 | --- | --- |
 | `appium:bundleId` | Bundle identifier of the app under test, for example `com.mycompany.myapp`. The capability value is calculated automatically if `app` is provided. If neither `app` or `bundleId` capability is provided then XCUITest driver starts from the Home screen. |
+| `appium:initialDeeplinkUrl` | A deeplink URL used to run either the application assigned to `appium:bundleId`, or the default application assigned to handle the particular deeplink protocol if `appium:bundleId` is not set. If provided in combination with `browserName=safari` then makes Safari to start with the given URL preloaded, which speeds up the session startup. The value of `appium:initialSafariUrl` capability is ignored in such case. An error is thrown on session init if either the value of the capability is not a valid URL, or XCTest was not able to associate it with any existing app, or the actual iOS version is below *16.4* |
```

Comment:
It makes sense.

Btw, with youtube as bundleId app and `initialDeeplinkUrl` did not open the given URL (just default youtube home screen) with ios 17, so the behavior could depend on OS versions as well, actually. No bundleId, and only `initialDeeplinkUrl` worked pretty well. So I wondered if we could address no-`appium:bundleId` as primary usage with this... (I mean dropping _the application assigned to appium:bundleId_ in the description)


With `browserName: safari` opened the app with `initialDeeplinkUrl` as same as `safariInitialUrl` this morning, so really not sure what affected the behavior, but safariInitialUrl/initialDeeplinkUrl could behave the same... Then, the user should switch the context properly to NATIVE_APP to interact with opened app with the URL.

So, I wondered if we could improve this documentation with:

- remove "the application assigned to `appium:bundleId`"
    - to reduce possible confusion what I mentioned the above's behavior diff
- add to suggest adjusting the `context` to NATIVE_APP for `initialDeeplinkUrl` and (`safariInitialUrl` as well..?) for the deeplink behavior with browserName=safari

Reply:
I try to not do preliminary optimizations and just observe how people are going to use it. We may always improve that afterwards based on the actual feedback.